test/unit/Cover/editCover.js

```diff
@@ -39,7 +39,7 @@ describe('editCover', function () {
 
     const expectedRefund = segment.amount
       .mul(targetPriceRatio)
-      .mul(segment.period)
+      .mul(segment.period - 1)
```

Comment:
As far as i understand it:

We use previousExpiration here to compute the refund
https://github.com/NexusMutual/smart-contracts/blob/8cc2511924d23f1d20563ef71063411ecf656db6/contracts/modules/cover/Cover.sol#L334


previousExpiration is computed here as lastSegment.start + lastSegment.period. 

https://github.com/NexusMutual/smart-contracts/blob/8cc2511924d23f1d20563ef71063411ecf656db6/contracts/modules/cover/Cover.sol#L212

With the new change to the solidity code, the start moment is 1 second in the past so the expire is 1 second in the past versus how it used to be.


Therefore in the formula

```        vars.refund =
          previousPoolAllocation.premiumInNXM
          * (allocationRequest.previousExpiration - block.timestamp) // remaining period
          / (allocationRequest.previousExpiration - allocationRequest.previousStart); // previous period```
          
          The numerator is smaller by 1.
          
          But the denominator is the same because both allocationRequest.previousExpiration  and allocationRequest.previousStart are the same.
          
          
          Therefore this -1 is right.

Reply:
Disregard this comment above.